website/scripts/index_search_content.js

```diff
@@ -1,3 +1,104 @@
-const { indexDocsContent } = require('@hashicorp/react-search/tools')
+require('dotenv').config()
+const fs = require('fs')
+const path = require('path')
+const {
+  indexContent,
+  getDocsSearchObject,
+} = require('@hashicorp/react-search/tools')
+const resolveNavData = require('../components/remote-plugin-docs/utils/resolve-nav-data')
+const fetchGithubFile = require('../components/remote-plugin-docs/utils/fetch-github-file')
 
-indexDocsContent()
+const GITHUB_API_TOKEN = process.env.CI_GITHUB_TOKEN
+
+// Run indexing
+indexContent({ getSearchObjects })
+
+async function getSearchObjects() {
+  // Resolve /docs, /guides, and /intro nav data, which
+  // corresponds to all the content we will actually render
+  // This avoids indexing non-rendered content, and partials.
+  // Fetch objects for `docs` content
+  async function fetchDocsObjects() {
+    const navFile = 'data/docs-nav-data.json'
+    const contentDir = 'content/docs'
+    const opts = {
+      remotePluginsFile: 'data/docs-remote-plugins.json',
+      githubToken: GITHUB_API_TOKEN,
+    }
+    const navData = await resolveNavData(navFile, contentDir, opts)
+    return await searchObjectsFromNavData(navData, 'docs')
+  }
+  // Fetch objects for `guides` content
+  async function fetchGuidesObjects() {
+    const navFile = 'data/guides-nav-data.json'
+    const contentDir = 'content/guides'
+    const navData = await resolveNavData(navFile, contentDir)
+    return await searchObjectsFromNavData(navData, 'guides')
+  }
+  // Fetch objects for `intro` content
+  async function fetchIntroObjects() {
+    const navFile = 'data/intro-nav-data.json'
+    const contentDir = 'content/intro'
+    const navData = await resolveNavData(navFile, contentDir)
+    return await searchObjectsFromNavData(navData, 'intro')
+  }
```

Comment:
Normally I would say this looks like it should be abstracted but in this scenario I don't think it matters that much really 😀